core/server/data/schema/schema.js

```diff
@@ -296,6 +296,12 @@ module.exports = {
         id: {type: 'string', maxlength: 24, nullable: false, primary: true},
         event: {type: 'string', maxlength: 50, nullable: false, validations: {isLowercase: true}},
         target_url: {type: 'string', maxlength: 2000, nullable: false},
+        name: {type: 'string', maxlength: 191, nullable: true},
+        secret: {type: 'string', maxlength: 191, nullable: true},
+        api_version: {type: 'string', maxlength: 50, nullable: false, defaultTo: 'v2'},
+        integration_id: {type: 'string', maxlength: 24, nullable: true},
+        last_triggered_at: {type: 'dateTime', nullable: true},
+        status: {type: 'string', maxlength: 191, nullable: true},
```

Comment:
Hmm maybe we could make two fields: `status` (50- failed or successful) and `reason` (50-reason of failure).
Otherwise if we want to show the user why the webhook has failed, we have to abuse the status field?
But not sure, don't want to block the PR from shipping.

Reply:
> why the webhook has failed, we have to abuse the status field?

I totally agree thats not nice, and was thinking of having separate, but I also felt separating is somehow leading to lot of extra fields which are all actually properties/data of `last_triggered_at`, and not webhooks as such, which doesn't seem intuitive. Even from the naming point of view, as we are following with `last_triggered_at`, thinking if `last_triggered_status` is more explicit than `status`, and similarly for naming of `reason` if we add that.

Reply:
It def should not be a blocker for shipping, but ideally the changes are small and we just need to decide on right pattern, so I think its better to do it right here than have another migration down the line :)

Reply:
I would've thought we'd store a HTTP "status" code in the status field, which we can use for both?

Reply:
Personally I would keep status available and use the explicit last_triggered_status / last_triggered_error field.

`status` seems very generic and I can imagine a future requirement where webhooks could have varying status’. Eg. Temporarily disabling some web hooks whilst you make some major site changes, or putting all webhooks into a “disabled” state after an import so that local development sites don’t start pinging production webhooks.

Reply:
^ This sounds perfect to me, will use HTTP status codes for `last_triggered_status` and also add `last_triggered_error` for any custom errors we might need to send. Default `status` can be set to `available`

Reply:
👍  It's also worth keeping in mind our initial webhook discussions, if I remember correctly we eventually want to make full logs available so this may be a temporary solution that's "good enough" for now. Full logs would allow Ghost(pro) users to debug sometimes-failing webhooks or to see a recent history of when a webhook was triggered. If we have that then `status` could be put in in a "failing" state if we see x many failed requests in y period.

Reply:
✌🏻

Reply:
> we eventually want to make full logs available so this may be a temporary solution that's "good enough" for now. 

For this I had assumed we'll eventually extend by adding a table like `webhook_trigger_logs`(or better name) which stores all triggers or n latest triggers for a webhook, along with its associated data, and then show it on UI pretty much like [Github](https://developer.github.com/webhooks/testing/#listing-recent-deliveries) does.
[[https://developer.github.com/assets/images/payload_request_tab.png]]